common/djangoapps/student/tests/test_email.py

```diff
@@ -196,85 +199,6 @@ def test_send_email_to_inactive_user(self, mock_log):
                 )
 
 
-@patch('student.views.management.render_to_string', Mock(side_effect=mock_render_to_string, autospec=True))
-@patch('django.contrib.auth.models.User.email_user')
-class ReactivationEmailTests(EmailTestMixin, CacheIsolationTestCase):
```

Comment:
Actually `compose_and_send_activation_email` has its own tests already. Do you suggest more tests to be added?

Reply:
Regarding `compose_and_send_activation_email` returning JSON, it is actually the removed `send_reactivation_email_for_user` which is returning JSON. I think your confusion is caused by github showing the + - code in a wrong order for `management.py` , because both `compose_and_send_activation_email` and `send_reactivation_email_for_user` contain similar code. (( `git diff` will show you the code in the wrong order too ))

Reply:
@shadinaif thanks for addressing the concerns. While the unit tests are covered. I think we still need to keep the integration tests of the old `send_reactivation_email_for_user`.

Some unit tests will be redundant and would be removed, but the integration tests are needed in my opinion.

Reply:
Unfortunately, `send_reactivation_email_for_user` didn't have tests like that. But `compose_and_send_activation_email` is already handled in many test directly:
`pytest common/djangoapps/student/tests/test_email.py::ActivationEmailTests::test_do_not_send_email_and_do_activate`
and indirectly:
`pytest common/djangoapps/student/tests/test_email.py::ActivationEmailTests::test_send_email_to_inactive_user`

I belive this is enough already. Or should I add more test?

Reply:
You're right, there were no integrations tests deleted.